File: Astronomy/Astronomy/View Controllers/ViewController.m
//
//  ViewController.m
//  Astronomy
//
//  Created by Christopher Aronson on 7/22/19.
//  Copyright © 2019 Christopher Aronson. All rights reserved.
//

#import "ViewController.h"
#import "KRCNetwork.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    KRCNetwork *call = [[KRCNetwork alloc] init];
    NSURL *url = [NSURL URLWithString:@"https://api.nasa.gov/mars-photos/api/v1/manifests/curiosity?api_key=bnqLeQQBboxLPczeBIjFzWTTYaNlGdCnVtJlGaJZ"];
    
    [call NetworkCallForURL:url completion:^(NSDictionary * _Nullable json, NSError * _Nullable error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            NSLog(@"%@", json);
        }
    }];
}


@end

File: Astronomy-objc/Astronomy-objc/ViewController.swift
//
//  ViewController.swift
//  Astronomy-objc
//
//  Created by Matthew Martindale on 8/5/20.
//  Copyright © 2020 Matthew Martindale. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let manifestController = FetchManifest()

    override func viewDidLoad() {
        super.viewDidLoad()

        let rover = "curiosity"
        manifestController.fetchRoverManifest(rover) { (rover, error) in
            guard error == nil else {
                print("Error fetching rover manifest in VC: \(error!)")
                return
            }
            if let rover = rover {
                self.manifestController.fetchPhotos(from: rover, onSol: 10) { (photos, error) in
                    print(photos?.first)
                }
            }
        }
    }
    

}
